moviepy/video/fx/resize.py

```diff
@@ -16,43 +14,105 @@ def resizer(pic, new_size):
             interpolation = cv2.INTER_AREA
         return cv2.resize(+pic.astype("uint8"), (lx, ly), interpolation=interpolation)
 
-    resizer.origin = "cv2"
+    return (resizer, [])
 
-except ImportError:
 
+def _get_PIL_resizer():
     try:
-        # TRY USING PIL/PILLOW AS RESIZER
         from PIL import Image
-        import numpy as np
+    except ImportError:
+        return (None, ["PIL not found (install 'Pillow')"])
 
-        def resizer(pic, new_size):
-            new_size = list(map(int, new_size))[::-1]
-            # shape = pic.shape
-            # if len(shape) == 3:
-            #     newshape = (new_size[0], new_size[1], shape[2])
-            # else:
-            #     newshape = (new_size[0], new_size[1])
+    import numpy as np
 
-            pil_img = Image.fromarray(pic)
-            resized_pil = pil_img.resize(new_size[::-1], Image.ANTIALIAS)
-            # arr = np.fromstring(resized_pil.tostring(), dtype='uint8')
-            # arr.reshape(newshape)
-            return np.array(resized_pil)
+    def resizer(pic, new_size):
+        new_size = list(map(int, new_size))[::-1]
+        # shape = pic.shape
+        # if len(shape) == 3:
+        #     newshape = (new_size[0], new_size[1], shape[2])
+        # else:
+        #     newshape = (new_size[0], new_size[1])
 
-        resizer.origin = "PIL"
+        pil_img = Image.fromarray(pic)
+        resized_pil = pil_img.resize(new_size[::-1], Image.ANTIALIAS)
+        # arr = np.fromstring(resized_pil.tostring(), dtype="uint8")
+        # arr.reshape(newshape)
+        return np.array(resized_pil)
 
+    return (resizer, [])
+
+
+def _get_scipy_resizer():
+    try:
+        from scipy.misc import imresize
     except ImportError:
-        # TRY USING SCIPY AS RESIZER
         try:
-            from scipy.misc import imresize
+            from scipy import __version__ as __scipy_version__
+        except ImportError:
+            return (None, ["Scipy not found (install 'scipy' or 'Pillow')"])
+
+        scipy_version_info = tuple(
+            int(num) for num in __scipy_version__.split(".") if num.isdigit()
+        )
+
+        # ``scipy.misc.imresize`` was removed in v1.3.0
+        if scipy_version_info >= (1, 3, 0):
+            return (
+                None,
+                [
+                    "scipy.misc.imresize not found (was deprecated in scipy v1.3.0,"
```

Comment:
I'm not sure that you understand the definition of "deprecated" (which I corrected in https://github.com/Zulko/moviepy/pull/1442/commits/c6922e52fd02369a995785f4ce14054a82383b89 as well). 

When a function is deprecated it is still available, and has not yet been removed. The deprecation is a signal that it is no longer being supported and will be removed in the future. According to https://docs.scipy.org/doc/scipy/reference/release.1.3.0.html#scipy-interpolate-changes, `imresize` was _deprecated_ in v1.0.0 and _removed_ in v1.3.0.